src/app/forms/suggestion-box/helper/terra-suggestion-box.helper.ts

```diff
@@ -0,0 +1,33 @@
+import { isNullOrUndefined } from 'util';
+import { TerraSuggestionBoxValueInterface } from '../data/terra-suggestion-box.interface';
+
+export class TerraSuggestionBoxHelper
+{
+
+    public static getSuggestionBoxEntryForCaption(caption:string,
+                                                  terraSuggestionBoxValues:Array<TerraSuggestionBoxValueInterface>,
+                                                  fallbackValue:TerraSuggestionBoxValueInterface):TerraSuggestionBoxValueInterface
+    {
+        if(isNullOrUndefined(terraSuggestionBoxValues))
+        {
+            return fallbackValue;
+        }
+        for(let value of terraSuggestionBoxValues)
+        {
+            if(value.value === caption)
+            {
+                return value;
+            }
+        }
+        return fallbackValue;
+    }
+
+    public static generateSuggestionBoxEntryFromCaption(caption:string):TerraSuggestionBoxValueInterface
```

Comment:
Bitte Parameter- und Funktionsnamen einheitlich benennen, d.h. entweder mit `Entry` oder `Value`. Vorschlag: Nutze immer `Value`, da auch das Interface (`TerraSuggestionBoxValueInterface`) so genannt ist.

Reply:
Das renaming in Entries ist für mich definitiv passender. Wie Pascal oben meinte heisst das Interface `Value` d.h. bei einen renaming müsstest du schauen, was deprecated gesetzt werden muss usw.
z.b. ein weg  müsste gehen, du machst ein zweites Interface mit den selben Attributen, welches Entry in den Namen nutzt und benutzt dies überall. Meines Wissens müsste dann trotzdem noch alles für die die  `TerraSuggestionBoxValueInterface` nutzten laufen.

natürlich würdest du im Selben Zuge das TerraSuggestionBoxValueInterface auf deprecated setzten.

Reply:
Wieso sollte ich irgendwelche neuen Interfaces anlegen? Man kann es umbenennen und gut ....